hack/api_transformer/main.go

```diff
@@ -349,6 +366,49 @@ func convertHideFromDocs(doc *ast.CommentGroup) {
 	}
 }
 
+func convertTabsToHeadings(doc *ast.CommentGroup) {
+	if doc == nil {
+		return
+	}
+
+	var newComments []*ast.Comment
+	inTabset := false
+
+	for _, comment := range doc.List {
+		text := comment.Text
+
+		if strings.Contains(text, "{{<tabset") {
+			inTabset = true
+			newComments = append(newComments, &ast.Comment{Text: "//"})
+			continue
+		}
+
+		if inTabset {
+			if strings.Contains(text, "{{<tab name=") {
+				// Extract the tab name and convert it to a heading
+				start := strings.Index(text, "name=\"") + len("name=\"")
+				end := strings.Index(text[start:], "\"") + start
+				heading := text[start:end]
+				newComments = append(newComments, &ast.Comment{Text: "// #### " + heading}) // #### Heading
+			} else if strings.Contains(text, "{{</tab>}}") || strings.Contains(text, "{{</tabset") {
+				// Skip these lines
+				if strings.Contains(text, "{{</tabset>}}") {
+					// Set to false after the last tabset
+					inTabset = false
+				}
+				// Add a newline to separate the headings
+				newComments = append(newComments, &ast.Comment{Text: "//"})
+			} else {
+				newComments = append(newComments, comment)
+			}
+		} else {
+			newComments = append(newComments, comment)
+		}
+	}
+
+	doc.List = newComments
```

Comment:
I think you can simplify this function a lot by replacing the individual items in `doc.List`  instead of building a new list and replacing the entire list.